### DISCUSSION TITLE
Quarto documents eval=FALSE inconsistent with RStudio

### DISCUSSION THREAD

**Discussion by renatovargas** [posit-dev/positron#4104] [Q&A] [answered]
I have a Quarto document where I instruct readers on how code is presented in the guide. In RStudio I have the parameters `{r eval=FALSE}` added to the header of the chunk, because it's a dummy function. In RStudio if I go down to any subsequent chunk and click on "Run All Chunks Above", that chunk will be correctly ignored. Note that I am not talking about `quarto render`ing the document, just running the calculations while working further down the document testing out calculations. In Positron, when clicking on the similar button "Run Above" from any subsequent chunk, this chunk will throw an error, because the dummy function doesn't exist, but it is evaluated when it shouldn't be.

![image](https://github.com/user-attachments/assets/688e7674-7ec4-43f6-8a90-a0b371699b7d)

That is, except if we move the parameter out of the curly brackets and into the other header format `#| eval: FALSE` like below. Then "Run Above" in subsequent chunks will correctly ignore it in Positron, but the problem is that "Run All Chunks Above" will no longer ignore it in RStudio as it should.

![image](https://github.com/user-attachments/assets/d199be38-20e3-4c0b-93c3-bc8621929460)

Related, but not granting another post is that when running "Run Above" and there are many chunks above, Positron will mess up the order of execution and it throws all kinds of errors when objects that haven't been created are called. This in documents where this functionality with "Run All Chunks Above" is working ok in RStudio. When clicking on "Run Cell" one by one in Positron, then no problems arise.

Thank you for this great product. It's coming out great. Love its flexibility.

Don't know if this is actually an issue. What do you think?

**Comment by juliasilge** [comment, marked as answer]
I may not be quite understanding your problem regarding "Run Above". It is true that in Positron comments like this will _not_ work:

````
```{r eval=FALSE}
x <- 1 + 1
x
```
````

Instead, you need to do this kind of hash pipe comment:

````
```{r}
#| eval: false
x <- 1 + 1
x
```
````

(Notice that it is `false`, not `FALSE`, since this is YAML and not R code.) If you have a lot of files to convert in this way, you may want to consider using `knitr::convert_chunk_header()`.

For me, this works correctly for "Run Above" in Positron and "Run All Chunks Above" in RStudio. Can you clarify what you are seeing?

> when running "Run Above" and there are many chunks above, Positron will mess up the order of execution and it throws all kinds of errors when objects that haven't been created are called

I suspect this bug is the same as #4068, and I will add this info there. Thank you for the report!

**Reply by renatovargas** [reply]
I appreciate your quick reply. I can confirm that `|# eval: false` with lowercase does work in both Positron and RStudio  when combined with "Run Above" or "Run All Chunks Above" and was confused because the ones below with uppercase `FALSE` from another chunk do work fine in both IDEs, strangely. 

```
#| warning: FALSE
#| message: FALSE
```
Some thoughts.

1. I actually like the `#|` version better, because it makes for cleaner code when you have to cram many parameters like figure identifiers and such and kept that one because it didn't work with my incorrect uppercase `FALSE`. 
2. However, it's a shame that parameters of the form `{r eval=false}` do not work in Positron with "Run Above", talking from a consistency standpoint, because those are what RStudio inserts by default when clicking on the chunk little gear options. If you are collaborating with people using either, that can be a hassle to have to change constantly. I now know better, but someone's bound to hit this wall too. Perhaps here it would be best to change RStudio's default behavior to `#|`  when clicking on the little gear chunk options. Note that we are discussing the IDE's convenient "Run Above" functionality, but rendering the document via `quarto render` will work just fine with inline parameters.
3.  I know it's another product, which you are probably not in charge of, but this confusion comes from Quarto's documentation, where sometimes [they use one type](https://quarto.org/docs/computations/r.html#chunk-options) and sometimes [the other type](https://quarto.org/docs/authoring/cross-references.html#code-listings) in examples, without telling you that mixing and matching inline/multiline types will throw an exception with a cryptic error message and there is only [one single obscure answer in the discussions page about it](https://github.com/quarto-dev/quarto-cli/discussions/6311).

Thank you for pointing me towards `knitr::convert_chunk_header()` as well, and for paying attention to the users. These are revolutionary products and it's a great time to be alive. Great work!!!!

**Reply by juliasilge** [reply]
Makes sense! Thanks for sharing your feedback. Just for clarity or anybody coming by later to read, using parameters like `{r eval=false}` will work if you are _rendering_ via knitr but not if you are using the "Run Above" or similar UI affordance, because that comes from the Quarto extension and that extension only knows about the hash pipe YAML comments AFAIK.

**Reply by cccfran** [reply]
I recently have the same dilemma too. Is putting both `{r eval=false}` and `#| eval: false` in Rmd still the best way for compatibility across RStudio and Positron? If so, is there a way to use `knitr::convert_chunk_header()` that preserves both formats?

I also find that the chunk label does not show properly with either `{r name}` or `#| label: name`; it just shows `(code cell)` in the outline.

**Reply by juliasilge** [reply]
Can you check what version of Positron and the Quarto extension you have? I do see `#| label: name` working in `.Rmd`:

<img width="1386" height="919" alt="Screenshot 2026-06-19 at 1 49 45 PM" src="https://github.com/user-attachments/assets/9017dbf0-9503-4412-8bbc-fce5285a8c7d" />

In fact, it works a little _too_ well and there is two of everything 😅 which we are tracking in https://github.com/posit-dev/positron/issues/13907.

I believe that the `#| eval: false` syntax will work correctly in `.Rmd` in RStudio:

<img width="1197" height="845" alt="Screenshot 2026-06-19 at 1 53 16 PM" src="https://github.com/user-attachments/assets/9855130a-3572-43d8-a33a-d6c48752fb00" />

**Reply by cccfran** [reply]
Thanks! I see the problem: I need `#| eval: false` instead of `#| eval = FALSE`. So when using `knitr::convert_chunk_header()`, set `format = "YAML"` solved my problem.
